### ISSUE TITLE
Klipper/Mainsail - Webcam unable to load with Crowsnest v4 camera-streamer & webrtc

### ISSUE THREAD

**Issue by Edigorin** [QuinnDamerell/OctoPrint-OctoEverywhere#42]
After changing to camera-streamer & webrtc option in Crowsnest v4, the webcam is unable to load in Octoeverwhere. Webcam work fine in Mainsail itself

**Comment by QuinnDamerell** [comment]
@Edigorin, thanks for letting me know! I haven't tried that setup yet. Is there a guide or something you followed to setup v4?

**Comment by Edigorin** [comment]
Here's the short explanation from the Mainsail Discord, in the announcement channel:

A short guide to updating Crowsnest to v4:
```bash
cd ~/crowsnest
make uninstall

# update Crowsnest via update manager

cd ~/crowsnest
sudo make install
```
To use the new streamer, you have to change the mode in `crowsnest.conf` from `mjepg`/`ustreamer` to `camera-streamer`. After Crowsnest restart, you have to change the mode in Mainsail webcam settings to `WebRTC (camera-streamer)` and change the streaming URL to `/webcam/webrtc`.

Here's the link to the post: https://discord.com/channels/758059413700345988/781284268654854154/1110660341693677650

**Comment by Edigorin** [comment]
Here's an updated guide on installing V4:
https://crowsnest.mainsail.xyz/faq/upgrade-from-v3-to-v4

**Comment by QuinnDamerell** [comment]
Thanks for the update! I just pushed an update to the plugin yesterday that fixes some of the issues, but the WebRTC stream in mainsail still won't work. I think for now, I might have to force mainsail to use the old streaming stack only when connecting via OctoEverywhere. I know what I need to do in-order to make WebRTC work, but it requires changes in the mainsail logic, which I'm just starting the conversations for now.

**Comment by QuinnDamerell** [comment]
I'm still working with the `camera-streamer` dev on a full solution here, hopefully it will be ready soon!

**Comment by QuinnDamerell** [comment]
Update, this is still on-going. I'm still waiting on the deployment of the new packages of `camera-streamer`.

**Comment by flow23** [comment]
Hi, is this still worked on?

**Comment by QuinnDamerell** [comment]
Hey! Yeah, I need to follow up with the crowsnest and camera streamer people. The OctoEverywhere logic is already done and rolled out, so whenever it's completed on their end, it should all work!

**Comment by arcaspo** [comment]
Hi, i am still encountering this issue but only on mobile, when i view the webcam through octoeverywhere on desktop it shows up fine but on mobile the stream doesn't work.

**Comment by QuinnDamerell** [comment]
I know it's been a while, but this is still in progress. The upgrade to the latest camera streamer seems to have stalled due to other projects out of my control, so I'm waiting with the rest of you. :)

**Comment by prettyyellowdog** [comment]
How's the progress on this?  It would also help to get the camera functionality up and going on the Creality K2, which also uses webrtc.

**Comment by QuinnDamerell** [comment]
I'm still working on this, but the K2 and Creaility Hi are different WebRTC setups, so they are their own monsters. :D

**Comment by NegussugeN** [comment]
Are there updates to this? I am experience the same issues as the OP, both with the camera-streamer/ WebRTC feed not loading either in Mainsail or within OctoApp when they are accessed remotely.
